stage1/rootfs/usr_from_kvm/kernel/patches/reboot.patch

```diff
@@ -0,0 +1,22 @@
+--- linux-4.0/kernel/reboot.c~	2015-04-12 18:12:50.000000000 -0400
++++ linux-4.0/kernel/reboot.c	2015-05-06 09:45:56.991894271 -0400
+@@ -309,7 +309,7 @@
+ 	 * halt when pm_power_off is not set do it the easy way.
+ 	 */
+ 	if ((cmd == LINUX_REBOOT_CMD_POWER_OFF) && !pm_power_off)
+-		cmd = LINUX_REBOOT_CMD_HALT;
++		cmd = LINUX_REBOOT_CMD_RESTART;
```

Comment:
Right, unless lkvm provides an ACPI target for S5 and then emulates the hardware, the kernel will just halt rather than doing anything else. The only straightforward alternative I can think of that might be acceptable upstream would be to add a kernel parameter to control this behaviour.